### PR TITLE
Tcm 4134 fts public api

### PR DESCRIPTION
**Pull request by andy-radchenko**
Preview link for branch:
https://api-reference.smartling.com/preview/tcm-4134-fts-public-api/#tag/File-Machine-Translation

### CODE REVIEW

File: spec/file_translation/fts_upload.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+x-paths:
+
+  file_upload:
+    post:
+      summary: Upload file
+      description: >-
+        This request allows you to upload a file, which later can be used for other actions (like MT).
+        Maximum file size is 200MB.
```
[the remark this answers is not in the document]
Comment (andy-radchenko) [reply spec/file_translation/fts_upload.yaml:8]: it's true for all file formats, regardless of file type. we do not have specific limits on per fileType basis

**Comment by gracecocco** [comment]
> @gracecocco jfyi - we are planning on adding functionality that goes beyond plain MT - AI related functionality - human review etc. Should we be creating an endpoints that reference MT alone? These may soon be deprecated apis

@klegins-smartling Is this more of a semantics question? Like should we call the endpoints "MT" something or something more generic?

File: spec/file_translation/fts_mt.yaml [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+        - lang: curl
+          source: >
+            curl -H "Authorization: Bearer $smartlingToken" 
+            https://api.smartling.com/file-translations-api/v2/accounts/$smartlingAccountId/files/$smartlingFtsFileUid/mt/$smartlingMtUid/locales/{$smartlingLocaleId}/file
+
+  mt_get_translated_files:
+    get:
+      summary: Download all MT file (all locales)
+      description: |
+        Download a ZIP archive with all MT for the requested file.
```
Comment (gracecocco) [inline spec/file_translation/fts_mt.yaml:247]: "Download a ZIP archive of translated files for all requested locales."

"Unlike the "Download translated file (single locale) endpoint..."

File: spec/file_translation/fts_mt.yaml [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+
+        When you upload a UTF-16 character encoded file, then /file/get requests
+        for that file will have a character encoding of UTF-16. All other
+        uploaded files will return with a character encoding of UTF-8.
+
+        You can always use the content-type header in the response of a file/get
+        request to determine the character encoding.
+        
+        Zip archive can only be downloaded if translation to all locales
+        has completed(overall state has to be in COMPLETED or CANCELLED states)
```
Comment (gracecocco) [inline spec/file_translation/fts_mt.yaml:265]: need a space here between "completed" and "(overall state...."

File: spec/file_translation/fts_mt.yaml [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+      x-code-samples:
+        - lang: curl
+          source: >
+            curl -H "Authorization: Bearer $smartlingToken" 
+            https://api.smartling.com/file-translations-api/v2/accounts/$smartlingAccountId/files/$smartlingFtsFileUid/mt/$smartlingMtUid/status
+
+
+  mt_cancel:
+    post:
+      summary: Cancel file MT
```
Comment (gracecocco) [inline spec/file_translation/fts_mt.yaml:132]: Cancel file translation

File: spec/file_translation/fts_language_detection.yaml [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+          $ref: '#/components/responses/Error500ResponseDefinition'
+      x-code-samples:
+        - lang: curl
+          source: >
+            curl -X POST -H "Authorization: Bearer $smartlingToken" 
+            https://api.smartling.com/file-translations-api/v2/accounts/$smartlingAccountId/files/$smartlingFtsFileUid/language-detection
+
+  language_detection_status:
+    get:
+      summary: Language detection progress
```
Comment (gracecocco) [inline spec/file_translation/fts_language_detection.yaml:61]: Get language detection progress

File: spec/file_translation/fts_language_detection.yaml [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+      description: |-
+        Fetch language detection progress/result.
```
Comment (gracecocco) [inline spec/file_translation/fts_language_detection.yaml:63]: Get progress for an ongoing source language detection action. Result of language detection is propagated to response as soon as state changes to COMPLETED.

**Review by gracecocco — COMMENTED** [review]
Suggested some naming changes for the endpoint names and reworded the descriptions. @klegins-smartling let's discuss a future proof name for the API in general?

File: spec/file_translation/fts_common.yaml [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+      type: object
+      properties:
+        key:
+          description: Error key
+          example: generalError
+          type: string
+        message:
+          description: Human readable description of the error
+          example: 'Sorry, something went wrong. We notified the responsible team. If you still need to contact us please provide this error code: jtcorrr3'
+          type: string
```
[the remark this answers is not in the document]
Comment (andy-radchenko) [reply spec/file_translation/fts_common.yaml:127]: This is an example of response. we're expected to give some real example here. If I dot that for errorId then it's logical to assume that I need to change also localeIds, states etc in the example... 
It's located in example section as well. It does not make much sense to me

File: spec/file_translation/fts_common.yaml [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+              type: array
+          required:
+            - code
+            - errors
+          type: object
+      required:
+        - response
+    FTSFileType:
+      type: String
+      description: FTS supported file types
```
[the remark this answers is not in the document]
Comment (andy-radchenko) [reply spec/file_translation/fts_common.yaml]: fixed

File: spec/file_translation/fts_language_detection.yaml [hunk 9]
```diff
@@ -0,0 +1,167 @@
+x-paths:
+
+  language_detection_action:
+    post:
+      summary: Detect file language
+      description: >-
+        Trigger detection of source language of earlier uploaded file
```
[the remark this answers is not in the document]
Comment (andy-radchenko) [reply spec/file_translation/fts_language_detection.yaml]: fixed

File: spec/file_translation/fts_upload.yaml [hunk 10]
```diff
@@ -0,0 +1,86 @@
+x-paths:
+
+  file_upload:
+    post:
+      summary: Upload file
+      description: >-
+        This request allows you to upload a file, which later can be used for other actions (like MT).
```
[the remark this answers is not in the document]
Comment (andy-radchenko) [reply spec/file_translation/fts_upload.yaml]: fixed

File: spec/file_translation/fts_mt.yaml (hunk 2 above)
Comment (andy-radchenko) [reply spec/file_translation/fts_mt.yaml]: fixed

File: spec/file_translation/fts_mt.yaml (hunk 3 above)
Comment (andy-radchenko) [reply spec/file_translation/fts_mt.yaml]: fixed

File: spec/file_translation/fts_language_detection.yaml (hunk 5 above)
Comment (andy-radchenko) [reply spec/file_translation/fts_language_detection.yaml]: fixed

File: spec/file_translation/fts_language_detection.yaml (hunk 6 above)
Comment (andy-radchenko) [reply spec/file_translation/fts_language_detection.yaml]: fixed

**Review by gracecocco — APPROVED** [review]
lgtm!

File: spec/file_translation/fts_common.yaml [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+              items:
+                $ref: '#/components/schemas/FTSError'
+              type: array
+          required:
+            - code
+            - errors
+          type: object
+      required:
+        - response
+    FTSFileType:
```
[the remark this answers is not in the document]
Comment (andy-radchenko) [reply spec/file_translation/fts_common.yaml]: fixed

**Pull request merged**